admin_interface/templatetags/admin_interface_tags.py

```diff
@@ -16,45 +17,43 @@
 register = template.Library()
 
 
-@register.simple_tag(takes_context=True)
-def get_admin_interface_languages(context):
+@register.inclusion_tag("admin_interface/language_chooser.html", takes_context=True)
+def admin_interface_language_chooser(context):
     if not settings.USE_I18N:
         # i18n disabled
         return None
     if len(settings.LANGUAGES) < 2:
         # less than 2 languages
         return None
+    if "django.middleware.locale.LocaleMiddleware" not in settings.MIDDLEWARE:
+        warnings.warn(
+            "Language chooser requires 'django.middleware.locale.LocaleMiddleware' "
+            "in your MIDDLEWARE to work.",
+            stacklevel=1,
```

Comment:
Isn’t 1 the default value? Is it correct here? (to determine: check what file and line of code is shown as the origin of the warning in the output)

Reply:
Yes it's the default. The linter was complaining it wants it explicitly.

Reply:
Two notes:
- that sounds like a bad linter, or strange configuration
- the point is that stacklevel=1 is very rarely useful, it’s more common to see stacklevel=2 so that the user code with the problem is shown in the traceback